packages/fxa-settings/src/pages/Signin/SigninTotpCode/container.tsx

```diff
@@ -121,30 +122,29 @@ export const SigninTotpCodeContainer = ({
         },
       });
 
-      // Check authentication code
-      if (result.data?.verifyTotp.success === true) {
-        // Attempt to finish any pending key stretching upgrade. We cannot finalize
-        // a key stretching upgrade until the session is verified, which the process
-        // can only be finished after the account has been verified on accounts that
-        // require totp.
-        if (signinState?.sessionToken) {
-          await tryFinalizeUpgrade(
-            signinState?.sessionToken,
-            sensitiveDataClient,
-            'signin-totp',
-            credentialStatus,
-            getWrappedKeys,
-            passwordChangeStart,
-            passwordChangeFinish
-          );
-        }
-
-        return { status: true };
+      // Check authentication
+      if (!result.data?.verifyTotp.success) {
+        return { error: AuthUiErrors.INVALID_TOTP_CODE };
       }
 
-      return { status: false };
+      // Attempt to finish any pending key stretching upgrade. We cannot finalize
+      // a key stretching upgrade until the session is verified, which the process
+      // can only be finished after the account has been verified on accounts that
+      // require totp.
+      if (signinState?.sessionToken) {
+        await tryFinalizeUpgrade(
+          signinState?.sessionToken,
+          sensitiveDataClient,
+          'signin-totp',
+          credentialStatus,
+          getWrappedKeys,
+          passwordChangeStart,
+          passwordChangeFinish
+        );
+      }
+      return { error: undefined };
```

Comment:
Probably worth adding to FXA-9757 or somewhere else, I notice we return `undefined` sometimes and `null` other times when there's no error. Would be nice to have it all consistent.